syntax = "proto3";

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "SharedCommunication.proto";

package communication.sec;

// The encrypted messages should be sent as bytes.
// If you see a field type is defined as byte, it indicates
// that the field holds an encrypted message in it.

service SecretCommunicationService {
  rpc getPublicKey(google.protobuf.Empty) returns (PublicKey);
  rpc createInvoice(InvoicingRequest) returns (InvoicingResponse);
}

message PublicKey {
  bytes publicKey = 1;
}

message BookingRequest {
  string ride_id = 1;
  optional int32 seatNumber = 2;
  BookingType bookingType = 3;
  InvoicingClerkType invoicingClerkType = 4;
  UnitPriceType unitPriceType = 5;
  bytes unitPrice = 6;
  bytes coefficient = 7;
  bytes discountRate = 8;
  optional bytes seatPrice = 9;
}

